client_test.go

```diff
@@ -208,3 +210,20 @@ func TestUseFstatChecked(t *testing.T) {
 	testFstatOption(t, UseFstat(true), true)
 	testFstatOption(t, UseFstat(false), false)
 }
+
+type sink struct{}
+
+func (*sink) Close() error                { return nil }
+func (*sink) Write(p []byte) (int, error) { return len(p), nil }
+
+func TestBrokenServer(t *testing.T) {
+	// Simulate a server that sends a single packet with zero length,
+	// then closes the connection. This used to cause a panic in the client.
+
+	packet := []byte{0, 0, 0, 0}
+	w := &sink{}
+
+	r := bytes.NewReader(packet[:])
```

Comment:
no need to replace here, packet is already a slice

Reply:
Holdover from a previous version, removed now.